mod-source-record-storage-server/src/main/java/org/folio/dao/impl/LBRecordDaoImpl.java

```diff
@@ -0,0 +1,252 @@
+    log.info("Attempting get by instance id: {}", sql);
+    return select(sql, tenantId);
+  }
+
+  @Override
+  public String getTableName() {
+    return RECORDS_TABLE_NAME;
+  }
+
+  @Override
+  public String getColumns() {
+    return ColumnBuilder
+      .of(ID_COLUMN_NAME)
+      .append(SNAPSHOT_ID_COLUMN_NAME)
+      .append(MATCHED_PROFILE_ID_COLUMN_NAME)
+      .append(MATCHED_ID_COLUMN_NAME)
+      .append(GENERATION_COLUMN_NAME)
+      .append(RECORD_TYPE_COLUMN_NAME)
+      .append(INSTANCE_ID_COLUMN_NAME)
+      .append(STATE_COLUMN_NAME)
+      .append(ORDER_IN_FILE_COLUMN_NAME)
+      .append(SUPPRESS_DISCOVERY_COLUMN_NAME)
+      .append(CREATED_BY_USER_ID_COLUMN_NAME)
+      .append(CREATED_DATE_COLUMN_NAME)
+      .append(UPDATED_BY_USER_ID_COLUMN_NAME)
+      .append(UPDATED_DATE_COLUMN_NAME)
+      .build();
+  }
+
+  @Override
+  public String getId(Record record) {
+    return record.getId();
+  }
+
+  @Override
+  protected JsonArray toParams(Record record, boolean generateIdIfNotExists) {
+    if (generateIdIfNotExists && StringUtils.isEmpty(record.getId())) {
+      record.setId(UUID.randomUUID().toString());
+    }
+    if (StringUtils.isEmpty(record.getMatchedId())) {
+      record.setMatchedId(record.getId());
+    }
+    JsonArray params = new JsonArray()
+      .add(record.getId())
+      .add(record.getSnapshotId())
+      .add(record.getMatchedProfileId())
+      .add(record.getMatchedId())
+      .add(record.getGeneration())
+      .add(record.getRecordType());
+    if (Objects.nonNull(record.getExternalIdsHolder())) {
+      params.add(record.getExternalIdsHolder().getInstanceId());
+    } else {
+      params.addNull();
+    }
+    params
+      .add(record.getState())
+      .add(record.getOrder());
+    if (Objects.nonNull(record.getAdditionalInfo())) {
+      params.add(record.getAdditionalInfo().getSuppressDiscovery());
+    } else {
+      params.addNull();
+    }
+    if (Objects.nonNull(record.getMetadata())) {
+      params.add(record.getMetadata().getCreatedByUserId());
+      if (Objects.nonNull(record.getMetadata().getCreatedDate())) {
+        params.add(DATE_FORMATTER.format(record.getMetadata().getCreatedDate()));
+      } else {
+        params.addNull();
+      }
+      params.add(record.getMetadata().getUpdatedByUserId());
+      if (Objects.nonNull(record.getMetadata().getUpdatedDate())) {
+        params.add(DATE_FORMATTER.format(record.getMetadata().getUpdatedDate()));
+      } else {
+        params.addNull();
+      }
+    } else {
+      params.addNull().addNull().addNull().addNull();
+    }
+    return params;
+  }
+
+  @Override
+  protected RecordCollection toCollection(ResultSet resultSet) {
+    return new RecordCollection()
+      .withRecords(resultSet.getRows().stream().map(this::toEntity).collect(Collectors.toList()))
+      .withTotalRecords(resultSet.getNumRows());
```

Comment:
@OleksiiKuzminov @TarasSpashchenko do we need to return a number of all records which conform filter condition in the "totalRecords" field to support pagination? As I understand resultSet.getRows().size() == resultSet.getNumRows(), and resultSet.getNumRows() depends not only on filter condition but on limit and offset as well.

Reply:
Correct. Good observation. Looks like we will need a count query.

Reply:
Would it be appropriate to not set this here and perform the count query at the service layer?

Reply:
Maybe it would be more straight forward to augment the queries with count response. I will update accordingly.

Reply:
Thanks William!

Reply:
Will this solution work for now? Should be performant enough with Postgres 9.2 improvements to count query. The only caveat is if no rows are returned the total is unknown, assigned -1 for now. This can be remedied by more advanced query or splitting count query from actual query.

Reply:
I think it will be better to improve it if it's possible.

Reply:
I have updated the queries to always return total count. If there are no actual result rows, there will be just a single row with the total count present. This can occur if limit is 0 or if offset is greater than the limit. I have updated logic to handle this result set and added tests accordingly.

Reply:
Thank you for the improvement!